### PR TITLE
[SRVOCF-438] Serverless functions - updates and additions

### PR DESCRIPTION
**Pull request by rh-max**
Version(s): 4.6 and 4.8+

Issue: https://issues.redhat.com/browse/SRVOCF-438

Link to docs preview:
https://51032--docspreview.netlify.app/openshift-enterprise/latest/serverless/functions/serverless-functions-setup.html
https://51032--docspreview.netlify.app/openshift-enterprise/latest/serverless/functions/serverless-functions-getting-started.html

Dev & QE review: complete

### CODE REVIEW

**Comment by rh-max** [comment]
@lance @jrangelramos Could you PTAL? Thanks.

File: modules/serverless-functions-podman-macos.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+----
+$ podman machine init --memory=8192 --cpus=2 --disk-size=20
+----
+
+. Start the podman machine, which serves the Docker API on a UNIX socket:
++
+[source,terminal]
+----
+$ podman machine start
+FIXME example output
```
Comment (lance) [inline modules/serverless-functions-podman-macos.adoc:25]: Needs fixing?

File: serverless/functions/serverless-developing-nodejs-functions.adoc [hunk 2]
```diff
@@ -6,9 +6,6 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-:FeatureName: {FunctionsProductName}
-include::snippets/technology-preview.adoc[leveloffset=+2]
```
Comment (lance) [inline serverless/functions/serverless-developing-nodejs-functions.adoc:10]: Same as above - should we have a warning for runtimes not yet GA?

File: serverless/functions/serverless-developing-python-functions.adoc [hunk 3]
```diff
@@ -6,9 +6,6 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-:FeatureName: {FunctionsProductName}
-include::snippets/technology-preview.adoc[leveloffset=+2]
```
Comment (lance) [inline serverless/functions/serverless-developing-python-functions.adoc:10]: Same regarding GA/TP

**Review by lance — COMMENTED** [review]
Overall this looks good - thank you! There are a couple of spots with FIXME still, and I have some questions about TP for the runtimes not yet in GA.

File: modules/serverless-create-func-kn.adoc [hunk 4]
```diff
@@ -22,23 +22,23 @@ Before you can build and deploy a function, you must create it by using the Knat
...
 ----
-Project path:  /home/user/demo/examplefunc
-Function name: examplefunc
-Runtime:       typescript
-Template:      events
-Writing events to /home/user/demo/examplefunc
+Created typescript function in /home/user/demo/examplefunc
 ----
 +
 ** Alternatively, you can specify a repository that contains a custom template.
```
Comment (jrangelramos) [inline modules/serverless-create-func-kn.adoc:44]: For the template example the output also has changed (just as the above change in L41).

So the lines 55 to 59 should be replaced by
```
Created node function in /home/user/demo/examplefunc
```

File: modules/serverless-create-func-kn.adoc [hunk 5]
```diff
@@ -22,23 +22,23 @@ Before you can build and deploy a function, you must create it by using the Knat
 ----
 $ kn func create -r <repository> -l <runtime> -t <template> <path>
 ----
-** Accepted runtime values include `node`, `go`, `python`, `quarkus`, and `typescript`.
-** Accepted template values include `http` and `events`.
+** Accepted runtime values include `quarkus`, `node`, `typescript`, `go`, `python`, `springboot`, and `rust`.
++
+:FeatureName: `node`, `typescript`, `go`, `python`, `springboot`, and `rust` function runtimes
```
[the remark this answers is not in the document]
Comment (jrangelramos) [reply modules/serverless-create-func-kn.adoc:27]: +1

File: modules/serverless-functions-podman-macos.adoc (hunk 1 above)
Comment (jrangelramos) [reply modules/serverless-functions-podman-macos.adoc:25]: The output is a little big. Will you include it all?

```
Starting machine "podman-machine-default"
Waiting for VM ...
Mounting volume... /Users/myuser:/Users/user

This machine is currently configured in rootless mode. If your containers
require root permissions (e.g. ports < 1024), or if you run into compatibility
issues with non-podman clients, you can switch using the following command: 

	podman machine set --rootful

API forwarding listening on: /Users/myuser/.local/share/containers/podman/machine/podman-machine-default/podman.sock

The system helper service is not installed; the default Docker API socket
address can't be used by podman. If you would like to install it run the
following commands:

	sudo /usr/local/Cellar/podman/4.2.1/bin/podman-mac-helper install
	podman machine stop; podman machine start

You can still connect Docker API clients by setting DOCKER_HOST using the
following command in your terminal session:

	export DOCKER_HOST='unix:///Users/myuser/.local/share/containers/podman/machine/podman-machine-default/podman.sock'

Machine "podman-machine-default" started successfully
```

File: modules/serverless-functions-podman-macos.adoc [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+[NOTE]
+====
+_FIXME change for macOS_ On most systems, this socket is located at `/run/user/$(id -u)/podman/podman.sock`.
+====
+
+. Establish the environment variable that is used to build a function:
++
+[source,terminal]
+----
+FIXME use path from the previous step
```
[the remark this answers is not in the document]
Comment (jrangelramos) [reply modules/serverless-functions-podman-macos.adoc:37]: This matched the output (User name on previous sample output can be changed to match "MyUser" here)

**Review by jrangelramos — COMMENTED** [review]
Hello @rh-max I just added a few comments from my part

File: serverless/functions/serverless-developing-go-functions.adoc [hunk 7]
```diff
@@ -6,9 +6,6 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-:FeatureName: {FunctionsProductName}
-include::snippets/technology-preview.adoc[leveloffset=+2]
```
[the remark this answers is not in the document]
Comment (rh-max) [reply serverless/functions/serverless-developing-go-functions.adoc:10]: @lance Good point! Returned the note everywhere but for Quarkus. With the wording:
`Python functions runtime is a Technology Preview feature only. Technology Preview features are not supported with Red Hat production service...`

etc for each runtime

File: modules/serverless-create-func-kn.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (rh-max) [reply modules/serverless-create-func-kn.adoc]: Evidently, this note doesn't work in the middle of a procedure. Moved it to the section introduction.

File: modules/serverless-functions-podman-macos.adoc (hunk 1 above)
Comment (rh-max) [reply modules/serverless-functions-podman-macos.adoc]: Thanks Jeff, included the relevant parts.

File: modules/serverless-functions-podman-macos.adoc [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+To use advanced container management features, you might want to use podman with {FunctionsProductName}. To do so on macOS, you need to start the podman machine and configure the Knative (`kn`) CLI to connect to it.
+
+.Procedure
+
+. Create the podman machine:
++
+[source,terminal]
+----
+$ podman machine init --memory=8192 --cpus=2 --disk-size=20
```
[the remark this answers is not in the document]
Comment (matejvasek) [reply modules/serverless-functions-podman-macos.adoc:17]: I think it's all right.

**Comment by rh-max** [comment]
Thanks Lance and Jeff for reviews, all implemented.

**Review by nalhadef — COMMENTED** [review]
Looks good. No changes for me.

**Comment by abrennan89** [comment]
RE 
image
can the feature be reworded or something so that it reads properly? "node, typescript, go, python, springboot, and rust function runtimes is a Technology Preview" reads strangely to me, because it's a list of things and then "is" rather than "are"

**Comment by abrennan89** [comment]
One more sort of minor thing, but it seems like https://51032--docspreview.netlify.app/openshift-enterprise/latest/serverless/functions/serverless-functions-getting-started.html#serverless-kn-func-invoke_serverless-functions-getting-started should be a procedure like the deployment section, rather than a reference command? This could be followed up in another PR / issue though but I think worth noting.

**Comment by rh-max** [comment]
@abrennan89 PTAL https://github.com/openshift/openshift-docs/pull/51032/commits/7e9dcd8f19ee26095cfcb452a21c3789ed4c70ec

**Pull request merged**